jwst/linearity/tests/test_linearity.py

```diff
@@ -0,0 +1,596 @@
+    assert(outfile.pixeldq[560, 550] == dqflags.pixel['HOT'])
+    assert(outfile.pixeldq[550, 560] == dqflags.pixel['DEAD'])
+    assert(outfile.pixeldq[500, 300] == 2049)
+
+
+def test_lin_subarray():
+    """Test that the pipeline properly extracts the subarray from the reference file.
+    put dq flags in specific pixels and make sure they match in the output subarray file"""
+
+    # create input data
+    # create model of data with 0 value array
+    ngroups = 50
+    ysize = 224
+    xsize = 288
+
+    # create the data and groupdq arrays
+    csize = (1, ngroups, ysize, xsize)
+    data = np.full(csize, 1.0)
+    pixeldq = np.zeros((ysize, xsize), dtype=int)
+    groupdq = np.zeros(csize, dtype=int)
+
+    # create a JWST datamodel for MIRI data
+    im = MIRIRampModel(data=data, pixeldq=pixeldq, groupdq=groupdq)
+
+    im.meta.instrument.name = 'MIRI'
+    im.meta.instrument.detector = 'MIRIMAGE'
+    im.meta.instrument.filter = 'F1500W'
+    im.meta.instrument.band = 'N/A'
+    im.meta.observation.date = '2016-06-01'
+    im.meta.observation.time = '00:00:00'
+    im.meta.exposure.type = 'MIR_IMAGE'
+    im.meta.subarray.name = 'MASK1550'
+    im.meta.subarray.xstart = 1
+    im.meta.subarray.xsize = xsize
+    im.meta.subarray.ystart = 467
+    im.meta.subarray.ysize = ysize
+
+    # Read in reference file
+
+    dq = np.zeros((1024, 1032), dtype=int)
+    numcoeffs = 3
+
+    # place dq flags in dq array that would be in subarray
+    # MASK1550 file has colstart=1, rowstart=467
+    dq[542, 100:105] = 1
+
+    ref_model = LinearityModel()
+    ref_model.data = np.zeros(shape=(numcoeffs, 1024, 1032), dtype=np.float32)
+    ref_model.dq = dq
+
+    ref_model.meta.instrument.name = 'MIRI'
+    ref_model.meta.instrument.detector = 'MIRIMAGE'
+    ref_model.meta.subarray.xstart = 1
+    ref_model.meta.subarray.xsize = 1032
+    ref_model.meta.subarray.ystart = 1
+    ref_model.meta.subarray.ysize = 1024
+
+    # run through pipeline
+    outfile = lincorr(im, ref_model)
+
+    # read dq array
+    outpixdq = outfile.pixeldq
+
+    # check for dq flag in pixeldq of subarray image
+    assert(outpixdq[76, 100] == 1)
+    assert(outpixdq[76, 104] == 1)
+
+
+def test_wave_f2100w():
+    """make sure correct filter (F2100W) reference file is called"""
+
+    # create input data
+    # create model of data with 0 value array
+    nints = 1
+    ngroups = 50
+    ysize = 1024
+    xsize = 1032
+
+    # create a JWST datamodel for MIRI data
+    im = make_rampmodel(nints, ngroups, ysize, xsize)
+
+    im.meta.instrument.detector = 'MIRIMAGE'
+    im.meta.instrument.filter = 'F2100W'
+    im.meta.instrument.band = 'N/A'
+    im.meta.exposure.type = 'MIR_IMAGE'
+
+    # set filter
+    filter = 'F2100W'
+
+    # run pipeline
+    outfile = LinearityStep.call(im)
+
+    # get linearity reference file from header
+    linfile = outfile.meta.ref_file.linearity.name
+    print('Linfile', linfile)
```

Comment:
More print statements (and lines 304, 311, 312, ...)

Reply:
can remove print statements